### PR TITLE
fix splashing error caused by nomodule script

### PR DESCRIPTION
**Pull request by jwst9**
- fix splashing error
image

- restore search function

- remove fixed scrollbar

image

image

### CODE REVIEW

File: client/src/Substack.tsx [hunk 1]
```diff
@@ -30,26 +29,38 @@ const Substack: React.FC = () => {
...
           newScript.type = script.type
         }
+
         if (script.innerHTML) {
           newScript.innerHTML = script.innerHTML
         }
+
+        if (script.noModule) {
+          newScript.noModule = true
+        }
```
Comment (jwst9) [inline client/src/Substack.tsx:56]: Need to keep `noModule` attribute for this script not working
image

File: client/src/Substack.tsx [hunk 2]
```diff
@@ -72,14 +83,6 @@ const Substack: React.FC = () => {
...
       }
 
-      const icons = Array.from(document.querySelectorAll("link[rel$='icon']"))
-
-      for (const icon of icons) {
-        const newIcon = icon.cloneNode()
-        icon.remove()
-        document.head.appendChild(newIcon)
-      }
-
```
Comment (jwst9) [inline client/src/Substack.tsx:82]: Redundant code. Favicon is showing correctly without it.

File: server/routes/index.ts [hunk 3]
```diff
@@ -125,13 +125,13 @@ router.get('/substack',
...
       const { substackDomain, substackUrl } = res.locals
-      const url = substackUrl || `https://${substackDomain}/${req.query.url}`
+      const url = `https://${substackUrl}` || `https://${substackDomain}/${req.query.url}`
       let headers, data
       if (SubstackCache[url]?.cacheTime !== undefined && SubstackCache[url].cacheTime + CacheLife > Date.now()) {
         headers = SubstackCache[url].headers
         data = SubstackCache[url].data
       } else {
-        const response = await AxiosBase.get(url)
+        const response = await AxiosBase.get(url, { headers: { cookie: `intro_popup_last_hidden_at=${new Date().toISOString()};` } })
```
Comment (jwst9) [inline server/routes/index.ts:134]: Adding this cookie will return response without intro popup rendered

File: client/src/substack.scss [hunk 4]
```diff
@@ -13,17 +13,11 @@
...
-//}
 .subscribe-footer{
   display: none!important;
 }
-.portable-archive.empty-list{
-  display: none!important;
-}
 .subscribe-iframe{
-  height: 100px;
+  display: none!important;
```
Comment (polymorpher) [inline client/src/substack.scss]: might need this back because we still need a way for people to subscribe. I will check locally

**Comment by polymorpher** [comment]
nice, works

**Review by polymorpher — APPROVED** [review]

**Comment by polymorpher** [comment]
merged by pushing commits to main. Github branch checking is stuck

**Pull request closed without merging**